archive_updater_test.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import subprocess
+import os
+import tempfile
+import shutil
+from archive_updater import update, fill_message_directory
+from unittest import mock
+
+class ArchiveUpdaterUpdateTest(unittest.TestCase):
+
+    @mock.patch.object(subprocess, 'Popen')
+    def test_update_success(self, mock_popen):
+        mock_popen.return_value.communicate.return_value = (b'updated', b'')
+        self.assertTrue(update(''))
+    
+    @mock.patch.object(subprocess, 'Popen')
+    def test_update_failure(self, mock_popen):
+        mock_popen.return_value.communicate.return_value = (b'failed to fetch', b'no internet connection')
+        self.assertFalse(update(''))
+
+class ArchiveUpdaterFillMessageDirectoryTest(unittest.TestCase):
+        
+    def setUp(self):
+        self.tmp_dir = tempfile.mkdtemp()
+    def tearDown(self):
+        shutil.rmtree(self.tmp_dir)
+        
+    @mock.patch.object(subprocess, 'Popen')
+    @mock.patch.object(subprocess, 'call')
+    def test_fill_message_directory_success(self, mock_call, mock_popen):
+        def writeFile(args, stdout):
+            if 'show' in args:
+                stdout.write('SUCCESS')
+                return
+            stdout.write('FAIL')
```

Comment:
something we can look at later once the test issue is debugged:

This func doesn't need to actually check the args.
We could do that with [assert_has_calls()](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.assert_has_calls) on the mock itself.

As is, should this fail, we'd find out.
But we wouldn't know what the offending args was that caused it to fail.
Would require changing the test code to find that out, as opposed to using assert_has_calls().

Reply:
I'm a little confused as to what this means, do you think I need to change anything right now? So I should use something like assert_has_calls(), but what in the mock am I asserting has been called? I guess I'm just not familiar with it.

Reply:
Currently we're relying on this function to validate that `git show` is called for each of the files

I.e. if we called something else, the file contents would be different below.

This function doesn't necessarily need to be the place where we do this validation.

```python
  def writeFile(args, stdout):
     stdout.write('called with: ' + ' '.join(args))
   # loop below then checks that 'show' is in each file, perhaps also that it has the right commit, etc.
```

Reply:
ah I see okay!